File: Java/PalindromeNumber.java
public class PalindromeNumber
{
	public boolean isPalindrome(int x)
	{
		int[] array = new int[11];
		int i=1,j=0;
		if(x==0)
		{
			return true;
		}
		else 
		{
			while(x!=0)
			{
				array[i++]=x%10;
				x=x/10;
			}
		}
		if(array[1]<0)
		{
			array[j++]=0;
		}
		i--;
		while(i-j>1)
		{
			if(array[i]!=array[j])
			{
				return false;
			}
			i--;
			j++;
		}
		if(i==j)
		{
			return true;
		}
		return true;
	}
}

//testing -2147447412

File: C/PalindromeNumber.c
#include <limits.h>
#include <stdbool.h>

int reversePositive(int x)
{
	int res = 0;
	while (x != 0)
	{
		if (res > INT_MAX / 10)
			return 0;
		res = res * 10 + x % 10;
		x /= 10;
	}
	return res;
}

bool isPalindrome(int x)
{
	if (x < 0)
		return false;
	if (x == 0)
		return true;
	return x == reversePositive(x);
}
